### PR TITLE
Browse: Advanced filter improvements

### PR DESCRIPTION
**Pull request by gerhardol**
Contributes to #7598

## Proposed changes

Several related improvements, could be split if there are discussions.
There are some relations to #9243 too (a combined branch exixts in my repo).

1.  FileHistory: Show grid graph for submodules and directories
The history was garbled for directories and submodules, as --parents was not set.

2. Move full history arguments to RevisionReader
Handle arguments also for path filter in FormBrowse etc so git-log arguments can be set

3. FormFilter: Show graph also with Git filters
Grid would be incorrect (would look as submodule in FileHistory) as --parents was not included with Git filters.
With this, you can follow the history for several files as well as wildcards.

4. FormFilter: Only show max count when user overridden
Max count was shown as always active in the filter, using the settings value.

5. Browse: Toolbar button for Advanced filter
Button to promote the filter.

The button should also show if the filter is active (Fugue: funnel--exclamation.png) and the filter in a tooltip.
The file paths should be up-datable from context menu in Diff/FileTree tabs.
Note especially that follow renames is not implemented for the file paths.
Maybe the contents in the advanced filter should be retained (maybe the checkboxes state too).

## Screenshots 

| item | Before | After
| --- | --- | ---
| 1. | image | image
| 3. (path filter:"**/Rev*.cs") | image | image
| 4. | image | image
| 5. | - | image
| 5. | - | image

## Test methodology 

Mostly UI changes, a few tests updated.

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by mstv — APPROVED** [review]
👍 Thank you

File: GitUI/CommandsDialogs/FormFileHistory.cs [hunk 1]
```diff
@@ -305,23 +305,7 @@ private void LoadFileHistory()
...
-                else
-                {
-                    // rename following disabled
-                    res.revision = " --parents";
-                }
-
-                if (AppSettings.FullHistoryInFileHistory)
-                {
-                    res.revision = string.Concat(" --full-history ", AppSettings.SimplifyMergesInFileHistory ? "--simplify-merges " : string.Empty, res.revision);
+                    res.revision += $" {FindRenamesAndCopiesOpts()}";
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply GitUI/CommandsDialogs/FormFileHistory.cs]: Also inverted the return in this method.
(Could not make clean fixup! commits so I added to the history.)

**Review by RussKie — APPROVED** [review]

File: GitUI/CommandsDialogs/FormFileHistory.cs [hunk 2]
```diff
@@ -241,90 +241,72 @@ private void LoadFileHistory()
...
-
-                    ObjectId currentObjectId = null;
-                    foreach (var line in lines.Select(GitModule.ReEncodeFileNameFromLossless))
-                    {
-                        if (string.IsNullOrEmpty(line))
-                        {
-                            // empty filename after sha
-                            continue;
-                        }
+                    return (revision: (string?)null, path: $" \"{fileName}\"");
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormFileHistory.cs:249]: Removed the temporary variable res, returned the values directly.

**Comment by gerhardol** [comment]
@msftbot merge in 1 day

**Pull request merged**